### PR TITLE
getAppsInTar

### LINKED ISSUE #32048: Apps: shipped - nonshipped mismatch
Running into this issue when preparin the doc PR: https://github.com/owncloud/documentation/pull/4276 (

### PR DESCRIPTION
**Pull request by mmattel**
## Description

This PR adds `\OC_App::getAppsInTar();` including a prefilled `shipped.json` file with all apps that have been shipped in the 10.1 tar file.
Having that method, we can identify easily which apps are included in the tar file enabling upcoming PR´s to differenciate between them and manually installed once.

## Related Issue

- Fixes https://github.com/owncloud/core/issues/32048

## Motivation and Context

Make apps included in tar and manually installed apps differenciable

## How Has This Been Tested?

```
$tar[] = \OC_App::getAppsInTar();
var_dump($tar);
(with a prefilled shipped.json)
array(1) {
  [0]=>
  array(20) {
    [0]=>
    string(8) "comments"
    [1]=>
    string(12) "configreport"
    [2]=>
    string(3) "dav"
    [3]=>
    string(10) "encryption"
    [4]=>
    string(8) "external"
    [5]=>
    string(20) "federatedfilesharing"
    [6]=>
    string(10) "federation"
    [7]=>
    string(5) "files"
    [8]=>
    string(14) "files_external"
    [9]=>
    string(13) "files_sharing"
    [10]=>
    string(14) "files_trashbin"
    [11]=>
    string(14) "files_versions"
    [12]=>
    string(17) "files_videoplayer"
    [13]=>
    string(14) "firstrunwizard"
    [14]=>
    string(6) "market"
    [15]=>
    string(13) "notifications"
    [16]=>
    string(16) "provisioning_api"
    [17]=>
    string(10) "systemtags"
    [18]=>
    string(18) "updatenotification"
    [19]=>
    string(13) "user_external"
  }
}

(with a standard shipped.json)
array(1) {
  [0]=>
  NULL
}
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [x] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

**Comment by phil-davis** [comment]
The past tense of the English verb "to tar" (as in, to tar the road - put bitumen on it) is "tarred".
How do we spell the past tense of "sort of acronyms" like Tape ARchive = `tar` that are accidentally English words with a different meaning? ;)
(and almost nobody writes a `tar` ball to tape any more, and why is it a tar ball?)

**Comment by mmattel** [comment]
:man_facepalming: thanks @phil-davis 
I corrected the wording and squashed the commits

File: core/shipped.json [hunk 1]
```diff
@@ -17,5 +17,27 @@
     "files_external",
     "dav",
     "federatedfilesharing"
+  ],
+  "appsInTar": [
```
Comment (PVince81) [inline core/shipped.json]: this list should rather be generated by the release script which compiles several apps together.

maybe we can reduce this list here to the ones included in the core repo, and later on the release script will append the additional ones as added for the community and enterprise releases

**Comment by PVince81** [comment]
also cc @tboerger @patrickjahns as the list in shipped.json could be useful in automated deployment scenarios

**Comment by patrickjahns** [comment]
@PVince81 
How would you imagine the shipped.json to be included so it would not fail the integrity check?

**Comment by PVince81** [comment]
@patrickjahns the file needs to be modified / extended by the release script before the signature is applied

**Comment by patrickjahns** [comment]
@PVince81 
We don't sign again for custom containers or tarballs - hence I ask how you envision this to be updated

**Comment by PVince81** [comment]
@patrickjahns I wasn't aware of custom tarballs.

hmmm, the enterprise one is built on top of the community one, which is already signed... so it would require re-signing the community one with the new file... not so good

**Comment by DeepDiver1975** [comment]
As discussed in https://github.com/owncloud/core/issues/35777 the master branch will from now on hold the ownCloud 10 codebase. 

This PR targetted ownCloud 11 which is postponed to a far distant future.

Because of that I'm closing this PR and kindly ask you to re-submit this PR in a few days.

Thanks a lot for your patience

**Pull request closed without merging**